Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue

```diff
@@ -0,0 +1,457 @@
+            minValue: helpers.withMessage(
+                `Dependent must be under the age of
+                        ${configStore.webConfig.maxDependentAge}.`,
+                (value: string) =>
+                    new DateWrapper(value).isAfter(minBirthdate.value)
+            ),
+            maxValue: helpers.withMessage("Invalid Date.", (value: string) =>
+                new DateWrapper(value).isBefore(new DateWrapper())
+            ),
+        },
+        PHN: {
+            required: helpers.withMessage("Valid PHN is required.", required),
+            minLength: helpers.withMessage(
+                "Valid PHN is required.",
+                minLength(12)
+            ),
+            validPersonalHealthNumber: helpers.withMessage(
+                "Personal Health Number must be valid.",
+                ValidationUtil.validatePhn
+            ),
+            isNew: helpers.withMessage(
+                "This dependent has already been added.",
+                () => !isDependentAlreadyAdded.value
+            ),
+        },
+    },
+    accepted: { isChecked: sameAs(true) },
+}));
+
+const v$ = useVuelidate(validations, { dependent, accepted });
+
+function addDependent(): void {
+    isLoading.value = true;
+    dependentService
+        .addDependent(userStore.hdid, {
+            ...dependent.value,
+            PHN: dependent.value.PHN.replace(/\D/g, ""),
+        })
+        .then(async () => {
+            errorType.value = null;
+
+            await nextTick();
+
+            hideDialog();
+            emit("handle-submit");
+        })
+        .catch((err: ResultError) => {
+            if (err.statusCode === 429) {
+                setTooManyRequestsError("addDependentDialog");
+            } else {
+                errorMessage.value = err.resultMessage;
+                errorType.value = err.actionCode ?? null;
+            }
+        })
+        .finally(() => (isLoading.value = false));
+}
+
+function clear(): void {
+    dependent.value = {
+        firstName: "",
+        lastName: "",
+        dateOfBirth: "",
+        PHN: "",
+    };
+    accepted.value = false;
+    errorMessage.value = "";
+    errorType.value = null;
+    v$.value.$reset();
+}
+
+function handleSubmit(): void {
+    v$.value.$reset();
+    addDependent();
+}
+
+function hideDialog(): void {
+    clear();
+    isVisible.value = false;
+}
+
+function setTooManyRequestsError(key: string): void {
+    errorStore.setTooManyRequestsError(key);
+}
+
+function touchDateOfBirth(): void {
+    v$.value.dependent.dateOfBirth.$touch();
+}
+
+watch(() => dependent.value.dateOfBirth, touchDateOfBirth);
+</script>
+
+<template>
+    <v-dialog
+        id="add-dependent-dialog"
+        v-model="isVisible"
+        data-testid="add-dependent-dialog"
+        scrollable
+        width="700px"
+        persistent
+        no-click-animation
+    >
+        <template #activator="slotProps">
```

Comment:
NO WORK, but need a decision.

I see there's some inconsistency in the implementations. We'll need to decide on a convention here.

I prefer `#activator="{ props: slotProps }"` for the definition, otherwise we'll need to consider something more "complete" if we want to get the entire value passed through like `#activator="slotObj"` and then consume as you have through `slotObj.props`